src/check.rs

```diff
@@ -101,35 +191,148 @@ fn check_http_url(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
             resp.into_synthetic_error().unwrap(),
         ))))
     } else if resp.ok() {
-        Ok(())
+        // the URL might contain a fragment, in that case we need to check if
+        // the fragment exists
+        match url.fragment() {
+            Some(fragment) => check_http_fragment(url, fragment),
+            None => Ok(()),
+        }
     } else {
         Err(CheckError::Http(Box::new(HttpError::UnexpectedStatus(
             url.clone(),
             resp,
         ))))
     }
 }
+cached_key_result! {
+    CHECK_HTTP: UnboundCache<(String, String), ()> = UnboundCache::new();
+    Key = { (url.to_string(), fragment.to_string()) };
+    fn check_http_fragment(url: &Url, fragment: &str) -> Result<(), CheckError> = {
+        let mut url = url.clone();
+        url.set_fragment(None);
+
+        debug!("Checking fragment {} of URL {}.", fragment, url.as_str());
+        let html = ureq::get(url.as_str()).call().into_string().unwrap(); // we're only called when the HEAD request returns OK status
```

Comment:
Rather than making two separate network calls, can you make a `get` to start and then pass the response into here? I don't really like having network IO inside a cached function, ideally this would be a pure function.

Reply:
Made sure that HTTP GET is only made if fragment is present.

Reply:
About doing IO in the cached function, I think that's the expensive part, I think it makes more sense to cache that than parsing and HashSet lookup. Though I can be persuaded otherwise. I did a version without `check_file_fragment` and `check_http_fragment`, i.e. only `is_fragment_available`, felt wrong.